airflow/hooks/bash.py

```diff
@@ -0,0 +1,88 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import signal
+from subprocess import PIPE, STDOUT, Popen
+from tempfile import TemporaryDirectory, gettempdir
+from typing import Dict, Optional
+
+from airflow import AirflowException
+from airflow.hooks.base import BaseHook
+
+
+class BashHook(BaseHook):
+    """Hook for running bash commands"""
+
+    def __init__(self) -> None:
+        self.sub_process = None
+        super().__init__()
+
+    def run_command(self, command, env: Optional[Dict[str, str]] = None, output_encoding: str = 'utf-8'):
```

Comment:
```suggestion
    def run_command(self, command; List[str], env: Optional[Dict[str, str]] = None, output_encoding: str = 'utf-8'):
```

We should _enforce_ command to be a list here -- it's more secure (otherwise shell escaping issues are a big risk) and should be the only option this supports